parsec_mock/parsec_mock.py

```diff
@@ -44,7 +45,17 @@ def run_test(test_folder, parsec_socket):
         connection, client_addr = sock.accept()
         try:
             received_data = connection.recv(4096)
-            b64_received_data = base64.b64encode(received_data).decode("ascii")
+            all_received_data = received_data
+            # Keep polling the connection until there's no more to read
+            # for 0.5s.
```

Comment:
Indeed, forgot to copy that over too

Reply:
Why do we need to set the connection timeout? The value, for example, might be not long enough if a CI VM is started on a heavy loaded host. Can we avoid using timeouts?

Reply:
From what I can tell, if there's no timeout then the `recv` will just hang until something is available to read. I'll try again next week without timeout.

Reply:
I've checked, without the timeout on the mock we eventually reach the timeout of the client. So we have to decide what's a sensible value here. The client has 60s.

Reply:
Parking this for now, we can discuss it in the weekly meeting and get consensus on what to do here

Reply:
With this approach we always try to read for the hard-coded timeout time.  Therefore if we set the timeout value too small we still might not read all the data. If we set it too large we will be wasting time waiting although we already received all the data.
  We can check if we can parse the received data and continue reading if we can't. For example:
  
  ```
      while True:
        print("Waiting for connections", flush=True)
        connection, client_addr = sock.accept()
        try:
            print("Connection received from {}".format(client_addr), flush=True)
            # Optionally reduce the default connection timeout
            # connection.settimeout(10)
            received_data = connection.recv(4096)
            all_received_data = bytearray()
            # Read the socket in cycle until we either
            # - receive a correct command or
            # - fail to receive any data
            while received_data:
                all_received_data += received_data
                b64_received_data = base64.b64encode(all_received_data).decode("ascii")
                if b64_received_data in test_cases:
                    (name, test_case) = test_cases[b64_received_data]
                    print("Received expected request for test case {}".format(name), flush=True)
                    bin_response = base64.b64decode(test_case.test_data.response)
                    connection.sendall(bin_response)
                    break

                print("Received unexpected request {}".format(b64_received_data), flush=True)
                print("Continue receiving", flush=True)
                received_data = connection.recv(4096)

            if not received_data:
                print("No correct command received. No more data to read", flush=True)
        finally:
            print("Closing connection", flush=True)
            connection.close()
```

Reply:
Also not likely because at least the header fields for body/auth length would have to be different!

Reply:
Ok, updated to the implementation from above. I tried to expand it a bit to check whether the received data is a correct prefix for any of the known test cases (and otherwise bomb out), but it didn't work and I didn't feel like debugging it.